examples/communities/test_qlouvain.py

```diff
@@ -0,0 +1,63 @@
+"""This module contains tests for the quantum louvain community detection example."""
+from dataclasses import asdict
+import pytest
+
+import qlouvain
+from graph_instances import random_fcs_graph
+from test_louvain import sanity_check_input
+
+
+@pytest.fixture()
+def graph_b(rng):
+    return random_fcs_graph(
+        250,
+        community_size=50,
+        mu=0.3,
+        average_degree=5,
+        rng=rng,
+    )
+
+
+def test_qlouvain_sg(graph_b, rng):
+    solver = qlouvain.QLouvainSG(graph_b, rng=rng)
+    sanity_check_input(solver.A)
+
+    solver.louvain()
+
+    assert asdict(solver.stats) == {
+        "classical_control_method_calls": pytest.approx(0),
+        "classical_actual_queries": pytest.approx(1430),
+        "classical_expected_queries": pytest.approx(1565.6747913382508),
+        "quantum_expected_classical_queries": pytest.approx(1634.4651285549912),
+        "quantum_expected_quantum_queries": pytest.approx(1153.524995809208),
+    }
+
+
+def test_simple_qlouvain_sg(graph_b, rng):
+    solver = qlouvain.SimpleQLouvainSG(graph_b, rng=rng)
+    sanity_check_input(solver.A)
+
+    solver.louvain()
+
+    assert asdict(solver.stats) == {
+        "classical_control_method_calls": pytest.approx(0),
+        "classical_actual_queries": pytest.approx(4812),
+        "classical_expected_queries": pytest.approx(4822.857182305446),
+        "quantum_expected_classical_queries": pytest.approx(5175.580403305713),
+        "quantum_expected_quantum_queries": pytest.approx(2412.747268476625),
+    }
+
+
+def test_edge_qlouvain(graph_b, rng):
```

Comment:
This test is really slow. Why?

Reply:
I marked the qlouvain tests as slow since they need to run on somewhat large graphs to produce reasonable results.
Otherwise the EdgeQ test should be faster after fixing the indexing issue below